scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2304,10 +2585,20 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
         accept[RightArrow]
         val start = in.tokenPos
         def end = in.prevTokenPos
-        blockStatSeq() match {
-          case List(q: Quasi) => q.become[Term.Quasi]
-          case List(term: Term) => term
-          case other => atPos(start, end)(Term.Block(other))
+        if (token.is[Indentation.Indent]) {
+          indented {
+            blockStatSeq() match {
```

Comment:
This part seems to be repeated, could extract it to a function?

Reply:
done